op_crates/fetch/11_streams.js

```diff
@@ -1,1443 +1,703 @@
-        );
-      }
-      return (
-        this[sym.closedPromise].promise ??
-          Promise.reject(new TypeError("Invalid reader."))
-      );
+    function closeAlgorithm() {
+      return transformStreamDefaultSinkCloseAlgorithm(stream);
     }
 
-    cancel(reason) {
-      if (!isReadableStreamDefaultReader(this)) {
-        return Promise.reject(
-          new TypeError("Invalid ReadableStreamDefaultReader."),
-        );
-      }
-      if (!this[sym.ownerReadableStream]) {
-        return Promise.reject(new TypeError("Invalid reader."));
-      }
-      return readableStreamReaderGenericCancel(this, reason);
+    stream[_writable] = createWritableStream(
+      startAlgorithm,
+      writeAlgorithm,
+      closeAlgorithm,
+      abortAlgorithm,
+      writableHighWaterMark,
+      writableSizeAlgorithm,
+    );
+
+    function pullAlgorithm() {
+      return transformStreamDefaultSourcePullAlgorithm(stream);
     }
 
-    read() {
-      if (!isReadableStreamDefaultReader(this)) {
-        return Promise.reject(
-          new TypeError("Invalid ReadableStreamDefaultReader."),
-        );
-      }
-      if (!this[sym.ownerReadableStream]) {
-        return Promise.reject(new TypeError("Invalid reader."));
-      }
-      return readableStreamDefaultReaderRead(this);
+    function cancelAlgorithm(reason) {
+      transformStreamErrorWritableAndUnblockWrite(stream, reason);
+      return resolvePromiseWith(undefined);
     }
 
-    releaseLock() {
-      if (!isReadableStreamDefaultReader(this)) {
-        throw new TypeError("Invalid ReadableStreamDefaultReader.");
-      }
-      if (this[sym.ownerReadableStream] === undefined) {
-        return;
-      }
-      if (this[sym.readRequests].length) {
-        throw new TypeError("Cannot release lock with pending read requests.");
-      }
-      readableStreamReaderGenericRelease(this);
+    stream[_readable] = createReadableStream(
+      startAlgorithm,
+      pullAlgorithm,
+      cancelAlgorithm,
+      readableHighWaterMark,
+      readableSizeAlgorithm,
+    );
+
+    stream[_backpressure] = stream[_backpressureChangePromise] = undefined;
+    transformStreamSetBackpressure(stream, true);
+    stream[_controller] = undefined;
+  }
+
+  /** @param {WritableStream} stream */
+  function initializeWritableStream(stream) {
+    stream[_state] = "writable";
+    stream[_storedError] = stream[_writer] = stream[_controller] =
+      stream[_inFlightWriteRequest] = stream[_closeRequest] =
+        stream[_inFlightCloseRequest] = stream[_pendingAbortRequest] =
+          undefined;
+    stream[_writeRequests] = [];
+    stream[_backpressure] = false;
+  }
+
+  /**
+   * @param {unknown} v 
+   * @returns {v is number}
+   */
+  function isNonNegativeNumber(v) {
+    if (typeof v !== "number") {
+      return false;
+    }
+    if (Number.isNaN(v)) {
+      return false;
+    }
+    if (v < 0) {
+      return false;
     }
+    return true;
```

Comment:
return 0 <= v

Reply:
Varying from the spec makes readability issues when checking you are aligned. The spec specifies the logic this way. 😢 For alignment purposes, I would prefer to follow the spec.